index.js

```diff
@@ -1,8 +1,9 @@
 function setState(newState) {
+  const listenersLength = this.listeners.length;
   this.state = { ...this.state, ...newState };
-  this.listeners.forEach((listener) => {
-    listener(this.state);
-  });
+  for (let i = 0; i < listenersLength; i++) {
+    this.listeners[i](this.state);
+  }
```

Comment:
There is a filter iterator below this. This should probably be changed to a `for` loop as well.

Reply:
Changed as well

Reply:
While we’re at it, I would propose ++i instead of i++